modules/imgproc/src/imgwarp.cpp

```diff
@@ -417,6 +417,403 @@ class resizeNNInvoker :
     resizeNNInvoker& operator=(const resizeNNInvoker&);
 };
 
+#if CV_AVX2
+class resizeNNInvokerAVX4 :
+    public ParallelLoopBody
+{
+public:
+    resizeNNInvokerAVX4(const Mat& _src, Mat &_dst, int *_x_ofs, int _pix_size4, double _ify) :
+        ParallelLoopBody(), src(_src), dst(_dst), x_ofs(_x_ofs), pix_size4(_pix_size4),
+        ify(_ify)
+    {
+    }
+
+#pragma optimization_parameter target_arch=AVX
+    virtual void operator() (const Range& range) const
+    {
+        Size ssize = src.size(), dsize = dst.size();
+        int y, x, pix_size = (int)src.elemSize();
+        int width = dsize.width;
+        int avxWidth = width - (width & 0x7);
+#if (defined WIN32 || defined _WIN32 || defined __CYGWIN__)
```

Comment:
OpenCV includes CV_DECL_ALIGNED() macro that should be used instead of such conditional branches